### DISCUSSION TITLE
Worker Error Recovery

### DISCUSSION THREAD

**Discussion by zwjohn** [taskforcesh/bullmq#506] [Q&A]
This may be a question rather than an issue, sorry I am new to BullMQ.
I am testing BullMQ on a single core server, and I noticed that if the worker does not return or throw error during the worker processing, then the worker listener won't fire or called again, all subsequence jobs for the queue will become stalled jobs.
That means, if the worker failed for some reason, everything will stop working after that, I had to restart the server to have the stalled jobs processed. Is this by design or how to avoid this behavior, so that worker is always available handling jobs, even previous job crashed due to some reason?

Thanks!

    worker = new Worker("taskQueue", async job => {
        //run logic

        console.log("taskQueue event triggered..", job.name, job.id);
//may error out during processing
      await processTask();

        return job.data;
    }, connection.getWorkerRedisConnection());

**Comment by zwjohn** [comment]
another question about Sandboxes Processors, it's a separated process, how do I see the console.log output? as it doesn't show the console out put in the main thread output

**Comment by manast** [comment]
> I am testing BullMQ on a single core server, and I noticed that if the worker does not return or throw error during the worker processing, then the worker listener won't fire or called again, all subsequence jobs for the queue will become stalled jobs.
That means, if the worker failed for some reason, everything will stop working after that, I had to restart the server to have the stalled jobs processed. Is this by design or how to avoid this behavior, so that worker is always available handling jobs, even previous job crashed due to some reason?

This is not the normal behaviour of BullMQ. Since the processor is an async function it must either complete or throw an exception. If you processor never ends, yes then the job will never end :).
